nni/retiarii/experiment.py

```diff
@@ -0,0 +1,162 @@
+import dataclasses
+import logging
+import time
+
+from dataclasses import dataclass
+from pathlib import Path
+from threading import Thread
+from typing import Any, List, Optional
+
+from ..experiment import Experiment, TrainingServiceConfig
+from ..experiment import launcher, rest
+from ..experiment.config.base import ConfigBase, PathLike
+from ..experiment.config import util
+from .utils import get_records
+from .integration import RetiariiAdvisor
+from .converter.graph_gen import convert_to_graph
+
+_logger = logging.getLogger(__name__)
+
+@dataclass(init=False)
+class RetiariiExeConfig(ConfigBase):
+    experiment_name: Optional[str] = None
+    search_space: Any = '' # TODO: remove
+    trial_command: str = 'python3 -m nni.retiarii.trial_entry'
+    trial_code_directory: PathLike = '.'
+    trial_concurrency: int
+    trial_gpu_number: int = 0
+    max_experiment_duration: Optional[str] = None
+    max_trial_number: Optional[int] = None
+    nni_manager_ip: Optional[str] = None
+    debug: bool = False
+    log_level: Optional[str] = None
+    experiment_working_directory: Optional[PathLike] = None
+    # remove configuration of tuner/assessor/advisor
+    training_service: TrainingServiceConfig
+
+    def __init__(self, training_service_platform: Optional[str] = None, **kwargs):
+        super().__init__(**kwargs)
+        if training_service_platform is not None:
+            assert 'training_service' not in kwargs
+            self.training_service = util.training_service_config_factory(training_service_platform)
+
+    def validate(self, initialized_tuner: bool = False) -> None:
+        super().validate()
+
+## End of public API ##
```

Comment:
I don't think this is the end of public API?

Reply:
removed